pkg/controller.v2/trainjob_controller.go

```diff
@@ -15,3 +15,46 @@ limitations under the License.
 */
 
 package controllerv2
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kubeflowv2 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1"
+)
+
+type TrainJobReconciler struct {
+	log      logr.Logger
+	client   client.Client
+	recorder record.EventRecorder
+}
+
+func NewTrainJobReconciler(client client.Client, recorder record.EventRecorder) *TrainJobReconciler {
+	return &TrainJobReconciler{
+		log:      ctrl.Log.WithName("trainjob-controller"),
+		client:   client,
+		recorder: recorder,
+	}
+}
+
+func (r *TrainJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var trainJob kubeflowv2.TrainJob
+	if err := r.client.Get(ctx, req.NamespacedName, &trainJob); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("trainJob", klog.KObj(&trainJob))
```

Comment:
What is the value to configure controller runtime logger with ` klog.KObj(&trainJob)` ?
Also, since we add logger to the TrainJob reconciler struct, why do you create a new logger here ?

Reply:
> What is the value to configure controller runtime logger with  klog.KObj(&trainJob) ?
Also, since we add logger to the TrainJob reconciler struct, why do you create a new logger here ?

This is not new logger, actually this logger is propagated from Reconciler and we reuse it. 
Here, we add information for which objects are reconciling.

Reply:
Indeed, we do not generate new logger, and not add a logger name (`WithName`) here. That it `WithValue` as you can see here.

Reply:
I see, that makes sense!